### PR TITLE
Provide an option not to delete the namespace flux is installed into.

### LINKED ISSUE #654: [Bug]: Provider should not delete namespace if it did not create it
### Describe the bug

Our deployment of flux looks like this
```
resource "kubernetes_namespace" "flux_system" {
  metadata {
    name = var.flux_namespace
  }
  ...<details>
}


resource "flux_bootstrap_git" "this" {
  namespace        = var.flux_namespace
  branch           = data.gitlab_project.cluster_state.default_branch
  components_extra = ["image-reflector-controller", "image-automation-controller"]
  ...<auth details>
}
```

We recently got an incident because flux decided to delete a namespace it didn't create upon destruction. I think the culprit is here: https://github.com/fluxcd/terraform-provider-flux/blob/main/internal/provider/resource_bootstrap_git.go#L652C77-L652C88

The terraform provider should attempt to delete the namespace only if it created it, not if it was created externally.

`flux uninstall` cli command provides an option to avoid this behavior: `--keep-namespace` 

### Steps to reproduce

As flux was stuck for a different reason those are the steps that I undertook. With the code above,
1. terraform apply
2. terraform state rm 'flux_bootstrap_git.this'
3. terraform apply --target='flux_bootstrap_git.this'

I believe the same could be reproduced by simply
1. terraform apply
2. terraform destroy --target='flux_bootstrap_git.this'

### Expected behavior

Namespace is not destroyed at any point

### Screenshots and recordings

_No response_

### Terraform and provider versions

```
Terraform v1.7.1
on linux_amd64
+ provider registry.terraform.io/fluxcd/flux v0.23.0
```

### Terraform provider configurations

```
provider "aws" {

  region = var.region

  default_tags {
    tags = var.default_tags
  }
  assume_role {
    role_arn = "<role>"
  }
}

provider "kubernetes" {
  experiments {
    manifest_resource = true
  }

  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}

provider "helm" {
  kubernetes {
    host                   = data.aws_eks_cluster.cluster.endpoint
    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
    token                  = data.aws_eks_cluster_auth.cluster.token
  }
}

provider "flux" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}

provider "gitlab" {
  token    = local.gitlab_admin_token
  base_url = var.gitlab_base_url
}
```

### flux_bootstrap_git resource

```
resource "flux_bootstrap_git" "this" {
  namespace        = var.namespace
  branch           = data.gitlab_project.cluster_state.default_branch
  components_extra = ["image-reflector-controller", "image-automation-controller"]
  depends_on = [gitlab_deploy_key.main]
  url        = "ssh://${local.formatted_ssh_url}"
  ssh = {
    username    = local.username
    private_key = local.key_pem
  }
}
```

### Flux version

0.23.0

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by swade1987**
## Description
It provides an option not to delete the namespace flux is installed into.

## Motivation and Context
Fixes #654 

## How has this been tested?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation
- [x] I have updated the documentation (if required) with `make docs`

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/fluxcd/terraform-provider-flux/blob/main/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

## Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by swade1987** [comment]
@dempo93 this should do it, right.

File: internal/provider/resource_bootstrap_git.go [hunk 1]
```diff
@@ -649,10 +656,16 @@ func (r bootstrapGitResource) Delete(ctx context.Context, req resource.DeleteReq
...
-		return
+
+	// Only remove namespace if not keeping it.
+	if data.KeepNamespace.ValueBool() {
+		tflog.Debug(ctx, fmt.Sprintf("The keep_namespace variable was set to true. Skipping removal of %s namespace.", data.Namespace), map[string]interface{}{})
+	} else {
+		err = uninstall.Namespace(ctx, log.NopLogger{}, kubeClient, data.Namespace.ValueString(), false)
+		if err != nil {
+			resp.Diagnostics.AddError("Unable to remove namespace", err.Error())
+			return
```
Comment (stefanprodan) [inline internal/provider/resource_bootstrap_git.go:667]: We need to remove the `return` from here and all the above functions. This is what we do in the CLI, maybe the controllers were deleted by hand, or the CRDs, or even the namespace. We should log the not found errors and continue with the cleanup.

File: internal/provider/resource_bootstrap_git.go [hunk 2]
```diff
@@ -631,37 +638,43 @@ func (r bootstrapGitResource) Delete(ctx context.Context, req resource.DeleteReq
 	kubeClient, err := r.prd.GetKubernetesClient()
 	if err != nil {
 		resp.Diagnostics.AddError("Kubernetes Client", err.Error())
+		tflog.Error(ctx, "Unable to get Kubernetes client", map[string]interface{}{})
 		return
 	}
 	// TODO: Uninstall fails when flux-system namespace does not exist
```
Comment (stefanprodan) [inline internal/provider/resource_bootstrap_git.go:644]: This TODO can be removed now

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @swade1987 🥇

**Pull request merged**